added pro vs oss to main (#1338)
semgrep/semgrep-docs: docs/usage-and-billing.mdx

---
slug: usage-and-billing
append_help_link: true
title: Usage and billing 
description: "Learn about pricing, tiers, and feature support for the following Semgrep Products: Semgrep OSS, Semgrep Code, and Semgrep Supply Chain."
tags:
  - Semgrep Cloud Platform
  - Semgrep OSS
  - Team & Enterprise Tier
hide_title: true
---

import MoreHelp from "/src/components/MoreHelp";

<ul id="tag__badge-list">
  {Object.entries(frontMatter)
    .filter((frontmatter) => frontmatter[0] === "tags")[0]
    .pop()
    .map((value) => (
      <li class="tag__badge-item">{value}</li>
    ))}
</ul>

<!-- # Pricing and billing -->

# Usage and billing 

Learn about usage computation and other aspects of your Semgrep licenses.

:::note
- This document is for **Semgrep Pro** users; Semgrep OSS Engine does not have any usage limit. Compare [<i class="fa-regular fa-file-lines"></i>  Semgrep Pro and Semgrep OSS](/semgrep-pro-vs-oss).
:::

## Usage 

All Semgrep Pro products are free under the **Team** tier for **10 monthly contributors**. These products include:

- Semgrep Cloud Platform
- Semgrep Code
- Semgrep Supply Chain
- Semgrep Secrets (beta)

> A **contributor** is someone who has made at least **one** commit to a Semgrep-scanned private repository within the last month, starting from the **date of license purchase** if a license was purchased, or the date of account creation, for accounts using Semgrep within usage limits.

Reviewer: I think we need a little more emphasis on private here - I missed it and below got confused when I didn't think we'd mentioned it.

Any type of Semgrep Pro scan counts towards the contributor total. This includes:

> - Scanning with any Semgrep Pro product (Code, Supply Chain, and so on.)

Reviewer: ```suggestion
- Scanning with any Semgrep Pro product (Code, Supply Chain, and so on).
```

- Full scans on a repository or partial scans on a pull or merge request.

### Usage computation

Contributors are calculated using `git log` over the **past 30 days** (a rolling interval), not the beginning of each month. The start date is either:

- The date of your license purchase.
- The date of account creation, if you and your team are within usage limits.

**Bots** and similar automations are excluded from this count.

### Exceeding the free usage limit

Semgrep scans stop when the usage limit is exceeded. You can resume scanning through the following:

* A one-time 30-day free trial that starts automatically when the usage limit is exceeded for the first time.
* By purchasing additional licenses.

> * By waiting for the next billing cycle.

Reviewer: The through/by combo here is kind of awkward. Maybe it would work with 

...resume scanning through:

* A one-time...
* Purchasing...
* Waiting...

?

> If the very first scan exceeds the usage limit, Semgrep still completes the first scan and a one-time 30-day free trial automatically starts. After the trial concludes, if scans are run on private repositories that exceed the usage limit, scans will not run until additional licenses are purchased.

Reviewer: ```suggestion
If the first scan exceeds the usage limit, Semgrep still completes the first scan and a one-time 30-day free trial automatically starts. After the trial concludes, if scans are attempted on private repositories that exceed the usage limit, scans will not run until additional licenses are purchased.
```
Not picky about the replacement for "attempted", but "if scans are run, scans will not run" is not great.

### Usage limit on public projects

Public projects have no limits on contributors.

### Semgrep CLI commands subject to usage limits

> The `semgrep scan` command is subject to the usage limit only if the scan is by a logged-in or authenticated contributor. Semgrep computes contributor counts for any logged-in scan command (for example, `semgrep ci`, `semgrep scan`, and so on) when the Pro Engine, Supply Chain, or Pro rules are used.

Reviewer: ```suggestion
The `semgrep scan` command is subject to the usage limit only if the scan is by a logged-in contributor. Semgrep computes contributor counts for any logged-in scan command (for example, `semgrep ci`, `semgrep scan`, and so on) when the Pro Engine, Supply Chain, or Pro rules are used.
```
Not sure what the extra "or authenticated" means here. If it has a distinct meaning, happy to brainstorm ideas for a different word :)

## Determine your plan needs

Within your team or organization, assess the number of **contributors**. Contributors are members of your organization that make commits. That determines the number of **licenses** needed for the plan purchase.

> For example, if a project has 4 unique contributors who create commits during the billing period while Semgrep is scanning their repositories, only 4 licenses are required even if the organization has a total of 10 members. If these unique contributors commit to many projects within the same organization, they are counted once, so no additional cost is charged.

Reviewer: ```suggestion
For example, if a project has 4 unique contributors who create commits during the billing period while Semgrep is scanning their repositories, only 4 licenses are required even if the organization has a total of 10 members. Contributors are counted only once even if they commit to many projects within the same organization, so no additional licenses are required.
```

:::info
You only need to buy licenses for contributors over 10. The 11th contributor and onwards are charged. For example, an engineering team with 20 contributing developers will only pay for 10 licenses.
:::

All members of the organization, regardless of contributor (license) status, have access to paid features for the chosen tier. This means that project managers and other non-programming roles can still view the Semgrep Cloud Platform dashboard.

### Single-product purchases

> Yes, you can buy a single product. Products can also be disabled from the [Settings page](https://semgrep.dev/orgs/-/settings).

Reviewer: ```suggestion
You can choose to purchase a single product. Products can also be disabled from the [Settings page](https://semgrep.dev/orgs/-/settings).
```

Reviewer: This yes seems weird since the title is not a question. I'm ambivalent about "choose to purchase" so feel free to ignore that. "You can buy a single product" just sounded a little abrupt.

> ### Different number of licenses per product

Reviewer: ```suggestion
### Number of licenses per product
```
I'd either make this general or say "Same" because it is required to be the same.

You must purchase an **equal number of licenses** for each product you intend to use. For example, you cannot purchase 4 licenses of Semgrep Supply Chain and 9 licenses of Semgrep Code. 

> ## Semgrep add-on reconciliation of licenses

Reviewer: ```suggestion
## Reconciliation of licenses and usage
```

> If the organization exceeds the number of purchased licenses, the organization will be charged based on the number of licenses that exceeded the purchased amount. The additional charge starts the month after the use of licenses exceeds the contracted amount.

Reviewer: I'm struggling to give a clearer alternative here but this does feel confusing. I think the thing that feels odd is that we're saying "you'll be charged based on how much you exceed", but it's not ultra clear why the excess and the original would be separate unless you get into details of the payment structure, which only happens after this

Author: I see what you mean. Tried to make it clear that the licenses are bought for a set period of time indicated in a contract.

Check in with your Semgrep Account Executive every **60 days** if you need more licenses than initially purchased.

### Example of license reconciliation

On January 21st, you purchased annual licenses for 50 developers of Semgrep Supply Chain’s Team tier ($40 per developer per month). The 21st of the month is the start date of the annual contract. In the following month, on February 28th, the number of used developer licenses exceeded the original purchased quantity by 20 users. This requires a contract adjustment.

Contract adjustment:

- Since the organization’s use exceeded the amount of purchased licenses on February 28th, the future date of March 21st is selected to align with the remaining months in the contract. There are 10 months remaining in the contract.
- The additional amount charged, the add-on cost, is $8,000 ($40 per developer per month x 10 months x 20 users).
- Resulting add-on cost: **$8,000**

## Upgrade your plan

To upgrade to the Semgrep Code **Team tier** through a **credit card**:

> 1. In the Settings page, select the Payment tab.

Reviewer: ```suggestion
1. In the Settings page, select the **Usage & billing** tab.
```

2. Select the number of developers to purchase licenses for.
3. Fill in your payment details.

> ![Screenshot of payment menu](/img/billing-and-pricing-payment.png)<br />

Reviewer 2: Can we update this screenshot?

Author: Thanks for the catch!

> To purchase seats for Semgrep Supply Chain or to upgrade to the **Enterprise tier**, please [contact us](https://semgrep.dev/contact-us).

Reviewer: Shouldn't this line mention Secrets too?
